src/card-list.js

```diff
@@ -1,28 +1,47 @@
 import React from 'react'
-import Card from './card.js'
+import NoCards from './no-cards.js'
+import Cards from './cards'
+import FilterDrop from './filter-dropdown.js'
 
-export default function CardList(props) {
-  return (
-    <React.Fragment>
-      {props.cards.length === 0 && <div className="text-center mt-5">
-        <h2>You have no cards saved</h2>
-        <a className="btn btn-primary mt-3" role="button" href="#new">
-        Make a Card
-        </a>
-      </div>}
-      {props.cards.length > 0 &&
-      <div className="container-fluid">
-        <div className="row justify-content-start">
-          {props.cards.map((card) => {
-            return <Card
-              id={card.cardId}
-              key={card.cardId}
-              remove={props.remove}
-              card={card}/>
-          })}
-        </div>
-      </div>
-      }
-    </React.Fragment>
-  )
+export default class CardList extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {show: 'none'}
+    this.sort = this.sort.bind(this)
+  }
+  sort(event) {
+    if (!event) {
+      this.setState({show: 'none'})
+    }
+    else {
+      this.setState({show: event.target.value})
+    }
+  }
+  render() {
+    const topics = []
```

Comment:
You shouldn't need this array. Filter already returns a filtered array. Maybe use reduce here instead, seeing as you don't really want an array of cards, but an array of topics.